patches/minecraft/net/minecraft/world/entity/vehicle/Boat.java.patch

```diff
@@ -1,5 +1,14 @@
 --- a/net/minecraft/world/entity/vehicle/Boat.java
 +++ b/net/minecraft/world/entity/vehicle/Boat.java
+@@ -434,7 +_,7 @@
+             for(int i2 = i1; i2 < j1; ++i2) {
+                blockpos$mutableblockpos.m_122178_(l1, k1, i2);
+                FluidState fluidstate = this.f_19853_.m_6425_(blockpos$mutableblockpos);
+-               if (fluidstate.m_76153_(FluidTags.f_13131_)) {
++               if (fluidstate.m_76153_(FluidTags.f_13131_) || fluidstate.canBoat(this)) {
```

Comment:
Given this is having to be changed anyways assuming that `canBoat` defaults to the isIn(Tag Water) check, it would be more readable if these patches just got changed to `if (fluidstate.canBoat(this)) {`. From a brief glance it looks like this isn't the default, though it should be made the default so any mods that call `canBoat` on the fluidstate get the correct result instead of also having to check if it is in the tag. That way modders can also define if they want a custom type of water that behaves as water for all purposes and is H2O but doesn't let boats float in it.